deeplearning4j-scaleout/spark/dl4j-spark-parameterserver/src/main/java/org/deeplearning4j/spark/parameterserver/pw/SharedTrainingWrapper.java

```diff
@@ -0,0 +1,237 @@
+package org.deeplearning4j.spark.parameterserver.pw;
+
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.exception.DL4JInvalidConfigException;
+import org.deeplearning4j.nn.api.Model;
+import org.deeplearning4j.nn.conf.WorkspaceMode;
+import org.deeplearning4j.optimize.solvers.accumulation.CudaGradientsAccumulator;
+import org.deeplearning4j.optimize.solvers.accumulation.MessageHandler;
+import org.deeplearning4j.parallelism.ParallelWrapper;
+import org.deeplearning4j.spark.parameterserver.conf.SharedTrainingConfiguration;
+import org.deeplearning4j.spark.parameterserver.iterators.VirtualDataSetIterator;
+import org.deeplearning4j.spark.parameterserver.iterators.VirtualIterator;
+import org.deeplearning4j.spark.parameterserver.iterators.VirtualMultiDataSetIterator;
+import org.deeplearning4j.spark.parameterserver.networking.SilentTrainingDriver;
+import org.deeplearning4j.spark.parameterserver.networking.WiredEncodingHandler;
+import org.deeplearning4j.spark.parameterserver.training.SharedTrainingResult;
+import org.deeplearning4j.spark.parameterserver.training.SharedTrainingWorker;
+import org.deeplearning4j.spark.parameterserver.util.BlockingObserver;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.dataset.api.MultiDataSet;
+import org.nd4j.parameterserver.distributed.VoidParameterServer;
+import org.nd4j.parameterserver.distributed.conf.VoidConfiguration;
+import org.nd4j.parameterserver.distributed.enums.TransportType;
+import org.nd4j.parameterserver.distributed.transport.MulticastTransport;
+import org.nd4j.parameterserver.distributed.transport.RoutedTransport;
+import org.nd4j.parameterserver.distributed.transport.Transport;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Observer;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This class maintains ParallelWrapper instance in Spark environment, and provides primitives for inter-executor
+ * communication during training over partitions.
```

Comment:
That's already happened, right there. SilentTrainerDriver provides connectivity with Aeron, and gradients are passed into GradientsAccumulator